src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -291,13 +291,19 @@ public async Task EndInvokeJSFromDotNet(long asyncCall, bool succeded, string ar
 
         public async Task DispatchEvent(string eventDescriptorJson, string eventArgs)
         {
+            AssertInitialized();
             RendererRegistryEventDispatcher.BrowserEventDescriptor eventDescriptor = null;
             try
             {
-                AssertInitialized();
-                eventDescriptor = ParseEventDescriptor(eventDescriptorJson);
-                if (eventDescriptor == null)
+                try
                 {
+                    eventDescriptor = JsonSerializer.Deserialize<RendererRegistryEventDispatcher.BrowserEventDescriptor>(
+                        eventDescriptorJson,
+                        JsonSerializerOptionsProvider.Options);
+                }
+                catch (Exception ex)
+                {
+                    Log.DispatchEventFailedToParseEventDescriptor(_logger, ex);
```

Comment:
Should we abort the circuit in this scenario (invalid incoming JSON)? This can't happen with a well-behaved client, right?

Reply:
I think we agreed in the past that if you send bad input we simply ignore it (log it Debug) and report an error, but that is it.

Reply:
I thought what we agreed was that we only log the bad input as debug, but thought the circuit should still die since there's no use case for invalid input that would never be sent by a well-behaved client.